src/helpers/helpers.ts

```diff
@@ -1,59 +1,100 @@
 import { wikibaseTimeToDateObject as toDateObject } from './wikibase_time_to_date_object.js'
+import type {
+  EntityId,
+  EntityPageTitle,
+  EntitySchemaId,
+  FormId,
+  Guid,
+  Hash,
+  ItemId,
+  LexemeId,
+  NonNestedEntityId,
+  NumericId,
+  PropertyClaimsId,
+  PropertyId,
+  RevisionId,
+  SenseId,
+} from '../types/entity.js'
 import type { Url } from '../types/options.js'
 
-export const isNumericId = (id: string): boolean => /^[1-9][0-9]*$/.test(id)
+export function isNumericId (id: unknown): id is NumericId {
```

Comment:
not sure, honestly if people just use TS, this all wouldn't be necessary.

Reply:
I use this kind of methods for input validation. So I don't know the type of something yet. TypeScript can't help in these cases as Types are only there at compile time but gone afterwards. So yeah, when only dealing with stuff within TypeScript it wouldn't be necessary. But as soon as some data from outside comes in (like from the wikibase API, files, whatever) it's good to ensure that the data I expect is that data. Or when I want to do something only for the Properties out of a bunch of different entities to only get the Properties out of it. So I do think that these methods are helpful even with TypeScript.

Reply:
guess the downside of accepting unknown is the missing typing checking on the input. passing in a number will return false and will not be found when compiling the typescript.

Reply:
I think using `string` as an input type and still using the `typeof id === 'string'` is the best of both worlds: If whatever passes in a number, object or whatever its still checked via the typeof and will return false. Also the TypeScript compiler checks for string inputs and will already be annoyed at compile time.

That way input validation is done and typings are useful.

Any other thoughts? If not I will change unknown → string in these methods.

Reply:
I just learned about type predicates thanks to this PR, amazing :grin:

Reply:
I changed it to `(id: string) => id is T` in 68d6d63